### PR TITLE
Remove `failure` crate

### PR DESCRIPTION
**Pull request by nshyrei**
Changes all usages of `failure` crate (https://crates.io/crates/failure) across the project into `thiserror` (https://crates.io/crates/thiserror) and `anyhow` (https://crates.io/crates/anyhow). This change fixes vulnerability introduced by the `failure` crate (https://rustsec.org/advisories/RUSTSEC-2020-0036)

### CODE REVIEW

**Review by Taowyoo — CHANGES_REQUESTED** [review]
The changes so far LGTM.

## Need to bump version number

But since this upgrade changes the function signatures, this will be a breaking change of all crates that are updated. **Which means you need to increase the minor version number of all crates you updated.**

## Concerns about changes and crate version before this breaking change

Before we merge this PR, I have to get clear about out release strategy about crates under `rust-sgx`.
If we want to have a version without this change but contains all changes before this PR and after last publish.
We need to publish new version of crates with new patch version if they only contain some version compatible changes.

Otherwise, we could just bump the minor versions in this PR.

**Comment by Taowyoo** [comment]
@raoulstrackx @ssavvides What's your idea about ^^^

File: intel-sgx/aesm-client/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "aesm-client"
-version = "0.5.4"
```
Comment (Taowyoo) [inline intel-sgx/aesm-client/Cargo.toml:3]: Hi @nshyrei, please check https://semver.org/
The version convention is `MAJOR.MINOR.PATCH`. Bump the PATCH version does not represent the breaking change.

You need to at least bump the MINOR version.
This applies to all your other version update, too.

Could you fix them, thank you.

**Review by Taowyoo — CHANGES_REQUESTED** [review]

File: intel-sgx/aesm-client/Cargo.toml (hunk 1 above)
Comment (Taowyoo) [reply intel-sgx/aesm-client/Cargo.toml:3]: You could use https://crates.io/crates/cargo-semver-checks to check if the version is file after you update them.

**Comment by nshyrei** [comment]
@Taowyoo Please review last 2 commits

File: intel-sgx/aesm-client/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+version = "0.6.4"
```
Comment (Taowyoo) [inline intel-sgx/aesm-client/Cargo.toml:3]:
```suggestion
version = "0.6.0"
```

File: intel-sgx/dcap-provider/Cargo.toml [hunk 3]
```diff
@@ -34,8 +34,8 @@ crate-type = ["cdylib"]
 
 [dependencies]
 # Project dependencies
-"dcap-ql" = { version = "0.3.0", path = "../dcap-ql", features = ["link"] }
-"report-test" = { version = "0.3.1", path = "../report-test" }
+"dcap-ql" = { version = "0.4.0", path = "../dcap-ql", features = ["link"] }
+"report-test" = { version = "0.4.1", path = "../report-test" }
```
Comment (Taowyoo) [inline intel-sgx/dcap-provider/Cargo.toml:38]:
```suggestion
"report-test" = { version = "0.4.0", path = "../report-test" }
```

File: intel-sgx/dcap-ql/Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "dcap-ql"
-version = "0.3.7"
+version = "0.4.7"
```
Comment (Taowyoo) [inline intel-sgx/dcap-ql/Cargo.toml:3]:
```suggestion
version = "0.4.0"
```

File: intel-sgx/dcap-ql/Cargo.toml [hunk 5]
```diff
@@ -54,7 +54,7 @@ yasna = { version = "0.3", features = ["num-bigint", "bit-vec"], optional = true
 
 [dev-dependencies]
 mbedtls = { version = ">=0.8.0, <0.10.0" }
-report-test = { version = "0.3.1", path = "../report-test" }
-sgxs = { version = "0.7.0", path = "../sgxs" }
+report-test = { version = "0.4.1", path = "../report-test" }
```
Comment (Taowyoo) [inline intel-sgx/dcap-ql/Cargo.toml:57]:
```suggestion
report-test = { version = "0.4.0", path = "../report-test" }
```

File: intel-sgx/dcap-retrieve-pckid/Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "dcap-retrieve-pckid"
-version = "0.1.3"
+version = "0.2.3"
```
Comment (Taowyoo) [inline intel-sgx/dcap-retrieve-pckid/Cargo.toml:3]:
```suggestion
version = "0.2.0"
```

File: intel-sgx/dcap-retrieve-pckid/Cargo.toml [hunk 7]
```diff
@@ -14,11 +14,11 @@ categories = ["command-line-utilities"]
 
 [dependencies]
 # Project dependencies
-"aesm-client" = { version = "0.5.0", path = "../aesm-client", features = ["sgxs"] }
-"dcap-ql" = { version = "0.3.0", path = "../dcap-ql", default-features = false }
-"report-test" = { version = "0.3.1", path = "../report-test" }
+"aesm-client" = { version = "0.6.0", path = "../aesm-client", features = ["sgxs"] }
+"dcap-ql" = { version = "0.4.0", path = "../dcap-ql", default-features = false }
+"report-test" = { version = "0.4.1", path = "../report-test" }
```
Comment (Taowyoo) [inline intel-sgx/dcap-retrieve-pckid/Cargo.toml:19]:
```suggestion
"report-test" = { version = "0.4.0", path = "../report-test" }
```

File: intel-sgx/enclave-runner/Cargo.toml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclave-runner"
-version = "0.5.2"
+version = "0.6.2"
```
Comment (Taowyoo) [inline intel-sgx/enclave-runner/Cargo.toml:3]:
```suggestion
version = "0.6.0"
```

File: intel-sgx/fortanix-sgx-tools/Cargo.toml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fortanix-sgx-tools"
-version = "0.5.1"
+version = "0.6.1"
```
Comment (Taowyoo) [inline intel-sgx/fortanix-sgx-tools/Cargo.toml:3]:
```suggestion
version = "0.6.0"
```

File: intel-sgx/report-test/Cargo.toml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "report-test"
-version = "0.3.3"
+version = "0.4.3"
```
Comment (Taowyoo) [inline intel-sgx/report-test/Cargo.toml:3]:
```suggestion
version = "0.4.0"
```

File: intel-sgx/sgxs-loaders/Cargo.toml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "sgxs-loaders"
-version = "0.3.3"
+version = "0.4.3"
```
Comment (Taowyoo) [inline intel-sgx/sgxs-loaders/Cargo.toml:3]:
```suggestion
version = "0.4.0"
```

File: intel-sgx/sgxs-tools/Cargo.toml [hunk 12]
```diff
@@ -25,12 +25,12 @@ path = "src/sgx_detect/main.rs"
...
-"sgxs" = { version = "0.7.0", path = "../sgxs", features = ["crypto-openssl"] }
-"sgxs-loaders" = { version = "0.3.0", path = "../sgxs-loaders" }
-"aesm-client" = { version = "0.5.0", path = "../aesm-client", features = ["sgxs"] }
+"sgxs" = { version = "0.8.0", path = "../sgxs", features = ["crypto-openssl"] }
+"sgxs-loaders" = { version = "0.4.0", path = "../sgxs-loaders" }
+"aesm-client" = { version = "0.6.0", path = "../aesm-client", features = ["sgxs"] }
 "sgx-isa" = { version = "0.4.0", path = "../sgx-isa" }
-"report-test" = { version = "0.3.1", path = "../report-test" }
-"enclave-runner" = { version = "0.5.0", path = "../enclave-runner" }
+"report-test" = { version = "0.4.1", path = "../report-test" }
```
Comment (Taowyoo) [inline intel-sgx/sgxs-tools/Cargo.toml:32]:
```suggestion
"report-test" = { version = "0.4.0", path = "../report-test" }
```

File: intel-sgx/sgxs/Cargo.toml [hunk 13]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "sgxs"
-version = "0.7.4"
+version = "0.8.4"
```
Comment (Taowyoo) [inline intel-sgx/sgxs/Cargo.toml:3]:
```suggestion
version = "0.8.0"
```

**Review by Taowyoo — CHANGES_REQUESTED** [review]
When using a new `MINOR` version, you should start with 0 for `MINOR`.
Note:
You could use this tool https://github.com/crate-ci/cargo-release to bump version in a more reliable way with command `argo release version minor` under root of each crate.

**Comment by nshyrei** [comment]
@Taowyoo Fixed, please review

**Review by Taowyoo — APPROVED** [review]
Good job! Thank you for working on this big upgrade!

**Pull request merged**